packages/style-engine/class-wp-style-engine.php

```diff
@@ -141,6 +144,44 @@ class WP_Style_Engine {
 				),
 			),
 		),
+		'effects'    => array(
+			'transition' => array(
+				'property_keys' => array(
+					'default' => 'transition',
+				),
+				'path'          => array( 'effects', 'transition' ),
+			),
+		),
+		'elements'   => array(
+			'link'   => array(
+				'path'     => array( 'elements', 'link' ),
+				'selector' => 'a',
+				'states'   => array(
```

Comment:
I was wondering whether to reduce verbosity we could auto generated these states.

Something like:

```
'states' => generate_state_defs('link', ['hover','focus','disabled'])
```

Maybe that's a bit unusual in a `const` though?

Another option is to generate them at runtime in ` generate_elements_styles`.

Reply:
Ok I went ahead and did this. Hope you don't mind 🙇

Reply:
> Ok I went ahead and dynamically generated them within the generate_elements_styles. It makes it a lot less verbose. Hope you don't mind 🙇

Good one! Yeah, the first pass was rough just to check if and how such a model would work.

> Can you think of any state selectors that we wouldn't want to map 1:1 from the state definition (i.e. hover, focus...etc) to the CSS pseudo selector equivalent (e.g. :hover, :focus .etc)?

I sat and stared at my keyboard for about 2 mins and couldn't come up with anything, which means there are probably zillions. But if nothing comes immediately to mind for both of us then maybe we're safe for now?

Reply:
🤞 we should be ok. If it does I guess we could make it optional to pass an options config where you can supply more fine grained control over the mapping. But for now a simple 1:1 seems to work.